docs/decisions/0020-redemption-attempt-record.rst

```diff
@@ -0,0 +1,60 @@
+0020 Redemption Attempt Record
+******************************
+
+Status
+======
+
+Proposed - April 2024
+
+Context
+=======
+The primary responsibilities of the ``subsidy_access_policy`` module are as follows:
+
+1. Act as a "read-orchtestrator", determining which learner-credit-based redemptions
+   are allowed under various conditions.
+2. Act as the entry-point for the redemption workflow, via the ``redeem`` view
+   and related ``SubsidyAccessPolicy.redeem()`` method. This responsibility depends on
+   the read-orchestrator responsibility above - the redemption flow must determine if redemption
+   is allowed before proceeding to request that the enterprise-subsidy service write
+   any ``transaction`` records.
+
+There is a general observability problem with the redemption workflow: outcomes that
+result in errors or non-redemption can only be inspected in a meaningful way from
+logs, or sometimes from the error details of the redemption HTTP response. This makes
+it difficult to debug the redemption workflow and gather metrics for engineering and
+non-engineering stakeholders alike.
+
+Decision
+========
+We'll introduce a new Django model, called ``RedemptionAttempt`` (or similar), to persist
+business logic outcomes and workflow results in the course of the redemption workflow. We should
+think of it as an audit-log type of persisted data - the presence or absence of a ``RedemptionAttempt``
+record should have no bearing on subsequent attempts of the redemption workflow.
+It should record the following types of data:
+
+1. The core inputs of the redeem workflow, including identifiers for the policy, subsidy,
+   catalog, learner, content, and optionally any related ``LearnerContentAssignment`` record.
+2. The state of the policy at the time of the redemption attempt - pointing at a historical
```

Comment:
Agreed, pointing at a historical `SubsidyAccessPolicy` makes sense versus needing to add specific model fields representing the state of the policy at the time.